Client/Frontend/Browser/UserScriptManager.swift

```diff
@@ -569,6 +597,42 @@ class UserScriptManager {
           $0.addUserScript(.init(source: providerJS, injectionTime: .atDocumentStart, forMainFrameOnly: true, in: .page))
         }
       }
+      
+      if WalletDebugFlags.isSolanaDappsEnabled,
+         let script = walletSolProviderScript,
+         tab?.isPrivate == false,
+         Preferences.Wallet.WalletType(rawValue: Preferences.Wallet.defaultSolWallet.value) == .brave {
+        $0.addUserScript(script)
+        if var providerJS = walletSolProviderScripts[.solana] {
+          providerJS = """
+            (function($Object) {
+              if (window.isSecureContext) {
```

Comment:
Yeah this is done to check that we're in a secure context before injecting.
The Ethereum provider script is [being injected the same way above](https://github.com/brave/brave-ios/blob/solana-dapps/Client/Frontend/Browser/UserScriptManager.swift#L590-L597), just lost behind the fold on Github on this PR. Additionally there is a [hasOnlySecureContent check](https://github.com/brave/brave-ios/blob/solana-dapps/Client/Wallet/EthereumProviderHelper.swift#L85-L89) in the `EthereumProviderHelper` (similar to `SolanaProviderHelper`).

Reply:
eth provider is also injected behind this. there was a [discussion](https://bravesoftware.slack.com/archives/C023VS4HJ6Q/p1655313974617269?thread_ts=1655216153.799099&cid=C023VS4HJ6Q) for this and result in [here](https://github.com/brave/brave-ios/blob/3f59b86e76503dc1b9497913d7fa37766ccefaa7/Client/Frontend/Browser/UserScriptManager.swift#L565)

Reply:
Thanks found it. I've been working on an EIP for this lately and had another wallet developer ask how this can be done on iOS. When I originally looked for it I didn't find this part of the code. I'll link them to this so they can see an example.